### PR TITLE
[Time series Informer] fix dtype of cumsum

### PR DESCRIPTION
**Pull request by kashif**
# What does this PR do?
Fix an issue when training Informer with FP16, the `cumsum` returns float32. 

See report here: https://discuss.huggingface.co/t/how-to-train-on-multiple-gpus-the-informer-model-for-time-series-forecasting/48984/3

### CODE REVIEW

File: src/transformers/models/informer/modeling_informer.py [hunk 1]
```diff
@@ -647,7 +647,7 @@ def forward(
         # calculate context for updating the attn_output, based on:
         # https://github.com/zhouhaoyi/Informer2020/blob/ac59c7447135473fb2aafeafe94395f884d5c7a5/models/attn.py#L74
         if self.is_decoder:
-            context = value_states.cumsum(dim=-2)
+            context = value_states.cumsum(dim=-2, dtype=torch.float32).to(value_states.dtype)
```
Comment (ArthurZucker) [inline src/transformers/models/informer/modeling_informer.py:650]:
```suggestion
            context = value_states.cumsum(dim=-2, dtype=value_states.dtype)
```

**Review by ArthurZucker — COMMENTED** [review]
Nice catch! Thanks for the fix. Is there a reason for float32 -> cast to the correct type instead of computing it in the correct dtype? (If so just add a small comment! )

File: src/transformers/models/informer/modeling_informer.py (hunk 1 above)
Comment (kashif) [reply src/transformers/models/informer/modeling_informer.py:650]: so I wanted the cumsum to be in float32 to avoid overflow etc. and then cast...

**Review by ArthurZucker — APPROVED** [review]
Thanks

**Pull request merged**